Lib/test/test_dis.py

```diff
@@ -1149,43 +1157,90 @@ def simple(): pass
 ]
 
 
-class InstructionTests(BytecodeTestCase):
+class InstructionTestCase(BytecodeTestCase):
+
+    def assertInstructionsEqual(self, instrs_1, instrs_2, /):
+        instrs_1 = [instr_1._replace(positions=None) for instr_1 in instrs_1]
+        instrs_2 = [instr_2._replace(positions=None) for instr_2 in instrs_2]
+        self.assertEqual(instrs_1, instrs_2)
+
+class InstructionTests(InstructionTestCase):
 
     def __init__(self, *args):
         super().__init__(*args)
         self.maxDiff = None
 
     def test_default_first_line(self):
         actual = dis.get_instructions(simple)
-        self.assertEqual(list(actual), expected_opinfo_simple)
+        self.assertInstructionsEqual(list(actual), expected_opinfo_simple)
 
     def test_first_line_set_to_None(self):
         actual = dis.get_instructions(simple, first_line=None)
-        self.assertEqual(list(actual), expected_opinfo_simple)
+        self.assertInstructionsEqual(list(actual), expected_opinfo_simple)
 
     def test_outer(self):
         actual = dis.get_instructions(outer, first_line=expected_outer_line)
-        self.assertEqual(list(actual), expected_opinfo_outer)
+        self.assertInstructionsEqual(list(actual), expected_opinfo_outer)
 
     def test_nested(self):
         with captured_stdout():
             f = outer()
         actual = dis.get_instructions(f, first_line=expected_f_line)
-        self.assertEqual(list(actual), expected_opinfo_f)
+        self.assertInstructionsEqual(list(actual), expected_opinfo_f)
 
     def test_doubly_nested(self):
         with captured_stdout():
             inner = outer()()
         actual = dis.get_instructions(inner, first_line=expected_inner_line)
-        self.assertEqual(list(actual), expected_opinfo_inner)
+        self.assertInstructionsEqual(list(actual), expected_opinfo_inner)
 
     def test_jumpy(self):
         actual = dis.get_instructions(jumpy, first_line=expected_jumpy_line)
-        self.assertEqual(list(actual), expected_opinfo_jumpy)
+        self.assertInstructionsEqual(list(actual), expected_opinfo_jumpy)
+
+    def test_co_positions(self):
```

Comment:
If the test suite is executed with the opt-out, this test is going to fail

Reply:
That is a valid concern :hmm: (do we have any more tests like this (?)). Let's introduce a decorator to `test.support`.

Reply:
This is an example I think https://github.com/python/cpython/blob/44f91fc802a2b71312e9abd9e8e9dcbf02e8216d/Lib/test/test_c_locale_coercion.py#L281-L283

But I think the config changes should all be done together probably depending on what order we merge the PRs in.

Reply:
Let's merge this and fix all tests in a separate (but quick) PR

Reply:
Rebasing for 0-indexed offsets...